x/gov/keeper/deposit.go

```diff
@@ -15,85 +17,38 @@ import (
 	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 )
 
-// GetDeposit gets the deposit of a specific depositor on a specific proposal
-func (keeper Keeper) GetDeposit(ctx context.Context, proposalID uint64, depositorAddr sdk.AccAddress) (deposit v1.Deposit, err error) {
-	store := keeper.storeService.OpenKVStore(ctx)
-	bz, err := store.Get(types.DepositKey(proposalID, depositorAddr))
-	if err != nil {
-		return deposit, err
-	}
-
-	if bz == nil {
-		return deposit, types.ErrDepositNotFound
-	}
-
-	err = keeper.cdc.Unmarshal(bz, &deposit)
-	if err != nil {
-		return deposit, err
-	}
-
-	return deposit, nil
-}
-
 // SetDeposit sets a Deposit to the gov store
 func (keeper Keeper) SetDeposit(ctx context.Context, deposit v1.Deposit) error {
-	store := keeper.storeService.OpenKVStore(ctx)
-	bz, err := keeper.cdc.Marshal(&deposit)
-	if err != nil {
-		return err
-	}
-
 	depositor, err := keeper.authKeeper.StringToBytes(deposit.Depositor)
 	if err != nil {
 		return err
 	}
-
-	return store.Set(types.DepositKey(deposit.ProposalId, depositor), bz)
-}
-
-// GetAllDeposits returns all the deposits from the store
-func (keeper Keeper) GetAllDeposits(ctx context.Context) (deposits v1.Deposits, err error) {
-	err = keeper.IterateAllDeposits(ctx, func(deposit v1.Deposit) error {
-		deposits = append(deposits, &deposit)
-		return nil
-	})
-
-	return
+	return keeper.Deposits.Set(ctx, collections.Join(deposit.ProposalId, sdk.AccAddress(depositor)), deposit)
 }
 
 // GetDeposits returns all the deposits of a proposal
 func (keeper Keeper) GetDeposits(ctx context.Context, proposalID uint64) (deposits v1.Deposits, err error) {
-	err = keeper.IterateDeposits(ctx, proposalID, func(deposit v1.Deposit) error {
+	err = keeper.IterateDeposits(ctx, proposalID, func(_ collections.Pair[uint64, sdk.AccAddress], deposit v1.Deposit) bool {
 		deposits = append(deposits, &deposit)
-		return nil
+		return false
 	})
 
 	return
 }
 
 // DeleteAndBurnDeposits deletes and burns all the deposits on a specific proposal.
 func (keeper Keeper) DeleteAndBurnDeposits(ctx context.Context, proposalID uint64) error {
-	store := keeper.storeService.OpenKVStore(ctx)
-
-	err := keeper.IterateDeposits(ctx, proposalID, func(deposit v1.Deposit) error {
-		err := keeper.bankKeeper.BurnCoins(ctx, types.ModuleName, deposit.Amount)
-		if err != nil {
-			return err
-		}
-
-		depositor, err := keeper.authKeeper.StringToBytes(deposit.Depositor)
-		if err != nil {
-			return err
-		}
-
-		err = store.Delete(types.DepositKey(proposalID, depositor))
-		if err != nil {
-			return err
-		}
-		return nil
+	coinsToBurn := sdk.NewCoins()
+	err := keeper.IterateDeposits(ctx, proposalID, func(key collections.Pair[uint64, sdk.AccAddress], deposit v1.Deposit) bool {
+		coinsToBurn = coinsToBurn.Add(deposit.Amount...)
```

Comment:
this is a behavioural change, for which we compute the total amount of coins to burn and then we burn them all in a single batch